File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/parser/adme/AdmeExecutiveAgencyInfoParser.java
package com.shmedo.configlibrary.iot.cmd.parser.adme;

import com.shmedo.configlibrary.iot.enums.IOTCommandType;
import com.shmedo.configlibrary.iot.interfaces.IOTResultParser;
import com.shmedo.configlibrary.iot.model.adme.AdmeExecutiveAgencyInfo;

import java.util.HashMap;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  12/29/20 <br/>
 * 描述：  解析ADME执行机构参数
 */
public class AdmeExecutiveAgencyInfoParser implements IOTResultParser<AdmeExecutiveAgencyInfo> {
    @Override
    public AdmeExecutiveAgencyInfo parse(String result) {
        AdmeExecutiveAgencyInfo info = new AdmeExecutiveAgencyInfo();
        try {
            String[] keyValues = result.split("&");
            HashMap<String, String> keyValueMap = new HashMap<>();

            for (String keyValue : keyValues) {
                String[] strs = keyValue.split("=");
                if (strs.length < 2) {
                    keyValueMap.put(strs[0], "");
                } else {
                    keyValueMap.put(strs[0], strs[1]);
                }
            }
            info.setMeastype(keyValueMap.getOrDefault("meastype", "NullKey"));
            info.setDatatype(keyValueMap.getOrDefault("datatype", "NullKey"));
            info.setDatareply(keyValueMap.getOrDefault("datareply", "NullKey"));
            info.setRoundwaitetime(keyValueMap.getOrDefault("roundwaitetime", "NullKey"));
            info.setRoundmeasinval(keyValueMap.getOrDefault("roundmeasinval", "NullKey"));
            info.setRoundmeasstart(keyValueMap.getOrDefault("roundmeasstart", "NullKey"));

            info.setDatainval(keyValueMap.getOrDefault("datainval", "NullKey"));
            info.setCompensatetime(keyValueMap.getOrDefault("compensatetime", "NullKey"));
            info.setDriveaddress(keyValueMap.getOrDefault("driveaddress", "NullKey"));
            info.setDownspeed(keyValueMap.getOrDefault("downspeed", "NullKey"));
            info.setInterdeep(keyValueMap.getOrDefault("interdeep", "NullKey"));
            info.setDownwaitetime(keyValueMap.getOrDefault("downwaitetime", "NullKey"));
            info.setUpspeed(keyValueMap.getOrDefault("upspeed", "NullKey"));
            info.setMeaspacing(keyValueMap.getOrDefault("measpacing", "NullKey"));
            info.setMeaintertime(keyValueMap.getOrDefault("meaintertime", "NullKey"));
            info.setMeabaseth(keyValueMap.getOrDefault("meabaseth", "NullKey"));
            info.setDwonblocked(keyValueMap.getOrDefault("dwonblocked", "NullKey"));
            info.setUntimenum(keyValueMap.getOrDefault("untimenum", "NullKey"));
            info.setDetectiontime(keyValueMap.getOrDefault("detectiontime", "NullKey"));
            info.setDetectionstart(keyValueMap.getOrDefault("detectionstart", "NullKey"));
            info.setDetectionend(keyValueMap.getOrDefault("detectionend", "NullKey"));
            info.setInterval_compensation(keyValueMap.getOrDefault("interval_compensation", "NullKey"));
            info.setInterval_fitting(keyValueMap.getOrDefault("interval_fitting", "NullKey"));
            info.setPoint_offset(keyValueMap.getOrDefault("point_offset", "NullKey"));

            return info;
        } catch (Exception ex) {
            ex.printStackTrace();
            return null;
        }
    }

    @Override
    public void validate(String result) {

    }

    @Override
    public IOTCommandType commandType() {
        return IOTCommandType.ADME_MD_GET_EXECUTIVE_AGENCY;
    }
}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/iot_cmd/parser/adme/AdmeExecutiveAgencyInfoParser.kt
package com.shmedo.lib.device.base.iot_cmd.parser.adme

import com.shmedo.lib.device.base.iot_cmd.enums.IOTCommandType
import com.shmedo.lib.device.base.iot_cmd.interfaces.IOTCommandParser
import com.shmedo.lib.device.base.iot_cmd.model.adme.AdmeExecutiveAgencyInfo
import com.shmedo.lib.device.base.iot_cmd.utils.IOTConstants

/**
 * 创建者：gonghe
 *
 * 创建时间：2023/12/19
 *
 * 描述： TODO
 *
 *
 */
class AdmeExecutiveAgencyInfoParser : IOTCommandParser<AdmeExecutiveAgencyInfo> {
    override fun parseInstance(keyValueMap: Map<String, String>): AdmeExecutiveAgencyInfo {
        return AdmeExecutiveAgencyInfo().apply {
            meastype = keyValueMap.getOrDefault("meastype", IOTConstants.NULL_KEY)
            datatype = keyValueMap.getOrDefault("datatype", IOTConstants.NULL_KEY)
            datareply = keyValueMap.getOrDefault("datareply", IOTConstants.NULL_KEY)
            roundwaitetime = keyValueMap.getOrDefault("roundwaitetime", IOTConstants.NULL_KEY)
            roundmeasinval = keyValueMap.getOrDefault("roundmeasinval", IOTConstants.NULL_KEY)
            updatedate = keyValueMap.getOrDefault("updatedate", IOTConstants.NULL_KEY)
            invalday = keyValueMap.getOrDefault("invalday", IOTConstants.NULL_KEY)
            roundmeasstart = keyValueMap.getOrDefault("roundmeasstart", IOTConstants.NULL_KEY)
            datainval = keyValueMap.getOrDefault("datainval", IOTConstants.NULL_KEY)
            compensatetime = keyValueMap.getOrDefault("compensatetime", IOTConstants.NULL_KEY)
            driveaddress = keyValueMap.getOrDefault("driveaddress", IOTConstants.NULL_KEY)
            downspeed = keyValueMap.getOrDefault("downspeed", IOTConstants.NULL_KEY)
            interdeep = keyValueMap.getOrDefault("interdeep", IOTConstants.NULL_KEY)
            downwaitetime = keyValueMap.getOrDefault("downwaitetime", IOTConstants.NULL_KEY)
            upspeed = keyValueMap.getOrDefault("upspeed", IOTConstants.NULL_KEY)
            pzspeed = keyValueMap.getOrDefault("pzspeed", IOTConstants.NULL_KEY)
            measpacing = keyValueMap.getOrDefault("measpacing", IOTConstants.NULL_KEY)
            meaintertime = keyValueMap.getOrDefault("meaintertime", IOTConstants.NULL_KEY)
            meabaseth = keyValueMap.getOrDefault("meabaseth", IOTConstants.NULL_KEY)
            dwonblocked = keyValueMap.getOrDefault("dwonblocked", IOTConstants.NULL_KEY)
            untimenum = keyValueMap.getOrDefault("untimenum", IOTConstants.NULL_KEY)
            detectiontime = keyValueMap.getOrDefault("detectiontime", IOTConstants.NULL_KEY)
            detectionstart = keyValueMap.getOrDefault("detectionstart", IOTConstants.NULL_KEY)
            detectionend = keyValueMap.getOrDefault("detectionend", IOTConstants.NULL_KEY)
            interval_compensation = keyValueMap.getOrDefault("interval_compensation", IOTConstants.NULL_KEY)
            bottom_safe_distance = keyValueMap.getOrDefault("bottom_safe_distance", IOTConstants.NULL_KEY)
            interval_fitting = keyValueMap.getOrDefault("interval_fitting", IOTConstants.NULL_KEY)
            point_offset = keyValueMap.getOrDefault("point_offset", IOTConstants.NULL_KEY)
        }
    }

    override fun commandType(): IOTCommandType = IOTCommandType.ADME_MD_GET_EXECUTIVE_AGENCY
}